cmake/modules/CheckLinkerDynamicRelocationFlags.cmake

```diff
@@ -0,0 +1,11 @@
+include(CheckCXXCompilerFlag)
+
+function(check_linker_supports_flag flag_to_test var_name)
+  set(CMAKE_REQUIRED_FLAGS "${flag_to_test}")
+  check_cxx_compiler_flag("" LINKER_SUPPORTS_NOEXTERN_PROTECTED_DATA)
+  if(LINKER_SUPPORTS_NOEXTERN_PROTECTED_DATA)
```

Comment:
Variable name `LINKER_SUPPORTS_NOEXTERN_PROTECTED_DATA ` is not relevant for here.
`result` would be sufficient I think.